overlord/state/notices.go

```diff
@@ -205,11 +205,20 @@ const (
 	// Recorded by "snap run" command when it is inhibited from running a
 	// a snap due an ongoing refresh.
 	SnapRunInhibitNotice NoticeType = "snap-run-inhibit"
+
+	// Recorded whenever a request prompt is created or resolved. The key for
+	// interfaces-requests-prompt notices is the request prompt ID.
+	RequestsPromptNotice NoticeType = "interfaces-requests-prompt"
```

Comment:
To follow the existing convention here, should this instead be "requests-prompt-notice"? And similarly for the one below?

Reply:
From these, I think the convention is not to have "-prompt" at the end of the notice type string:https://github.com/canonical/snapd/pull/14272/files#diff-4fc1b33549a08f2c1e8c5ae6ebdd8a0b859cc5e93b7208bc6656e02c208a9183R193-R211

It's a good question whether these should simply be "request-prompt" and "request-rule" instead of "interfaces-requests-prompt" and "interfaces-requests-rule-update". These changes are actually from #14269 though, so I'd rather address them there :)

Reply:
would that change be backwards compatible / would we be able to pre-emptively request the newer name as well without anything breaking? Currently we have to specify the existing `interfaces-requests-prompts` notice type in our request to long poll for notices.

Reply:
We'll leave the `interfaces-requests-prompt` and `interfaces-requests-rule-update` names unchanged.

For the record though, clients should be able to poll for both old and new names, if we ever do want to change something:

```
/v2/notices?types=interfaces-requests-prompt&types=interfaces-requests-prompt-update
```

For example. Snapd will ignore the unrecognized names and return any notices for the types which are recognized (and allowed by the connected interfaces).